lib/Decl.cpp

```diff
@@ -286,31 +286,21 @@ llvm::json::Object ValueDecl::SerializeToJSON(const llvm::DataLayout &dl) {
 }
 
 // Create a Function Declaration from llvm::Function
-FunctionDecl FunctionDecl::Create(const llvm::Function &func, const llvm::Module &mdl) {
-  const llvm::DataLayout dl = mdl.getDataLayout();
-  const remill::Arch* arch = remill::Arch::GetModuleArch(mdl);
+FunctionDecl FunctionDecl::Create(const llvm::Function &func,
+                                  const llvm::Module &mdl) {
+  const llvm::DataLayout& dl = mdl.getDataLayout();
+  const remill::Arch *arch = remill::Arch::GetModuleArch(mdl);
 
   FunctionDecl decl;
   decl.type = func.getFunctionType();
-  
   decl.name = func.getName().data();
   decl.demangled_name = llvm::demangle(func.getName().data());
   decl.dl = &dl;
 
   // Try to guess the parameter and return value register allocation by looking
   // at the calling convention of the function
-  llvm::CallingConv::ID cc_id = func.getCallingConv();
```

Comment:
A function might specify a calling convention, which might override the default. I think the `func.getCallingConv()` value should be passed in as a kind of "hint". If the value is `llvm::CallingConv::C` (this is the default value), then you do the arch+os-specifc heuristics, but if it's a non-default value, then use it.